### PR TITLE
Split "testing policies" section

### LINKED ISSUE #21: Documentation structure
Creating this issue to discuss the big picture structure of the docs:

1. **Introduction**
1. **Quick start**
1. **Writing policies**
    1. **Introduction**: here we could talk a bit about what a language needs to have to write policies with it, the waPC contract (functions, arguments and return structure)
    1. **Supported languages**
        3.2.1. **Rust**
        3.2.2. **TinyGo**: _characteristics, limitations as opposed to Go, and why Go is not an option right now_
        3.2.3. **AssemblyScript**: _characteristics, limitations_
1. **Distributing policies**
1. **Testing policies**
    1. **While developing** a policy
        1. _Use unit tests, integration... whatever you are used to_
        1. _Use the policy-testdrive project for offline tests: how you can run tests that take different settings, and requests checking the response of your policy on a Wasm build of the code you are developing_
    1. **Before deploying** a policy
        1. Use the policy-testdrive project for offline tests: _how you can run tests to understand how it would impact your cluster before the real deployment of your policy coming from a registry/HTTP server_

### PR DESCRIPTION
**Pull request by flavio**
Split the testing document into two dedicated chapters.

Also, shuffle the topics around inside of the SUMMARY

This fixes #21

### CODE REVIEW

File: src/testing-policies/01-intro.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Testing Policies
+
+This section covers the topic of testing Kubewarden Policies. There are two possible
+personas interested in testing policies:
+
+  * As a policy author: you're writing a Kubewarden Policy and you want to ensure
+    your code behaves the way you expect.
+  * As an end user: you found a Kubewarden Policy and you want to tune/test the policy
+    settings before deploying it, maybe you want to keep testing these settings
+    inside of your CI/CD pipelines,...
```
Comment (ereslibre) [inline src/testing-policies/01-intro.md:10]: Is `,...` intended?

File: src/testing-policies/03-cluster-operators.md [hunk 2]
```diff
@@ -237,34 +224,84 @@ As a matter of fact, the Pod using a custom `nginx` profile gets rejected by
...
 Settings validation result: SettingsValidationResponse { valid: true, message: None }
 Policy evaluation results:
 ValidationResponse { uid: "", allowed: true, patch_type: None, patch: None, status: None }
 ```
 
-## Wrapping up
+## Automation
+
+All these steps shown above can be automated using
+[bats](https://github.com/sstephenson/bats).
```
Comment (ereslibre) [inline src/testing-policies/03-cluster-operators.md:254]: Isn't this the unmaintained one? https://github.com/sstephenson/bats/issues/150 vs https://github.com/bats-core/bats-core

**Review by ereslibre — CHANGES_REQUESTED** [review]
LGTM, a couple of comments.

File: src/testing-policies/01-intro.md (hunk 1 above)
Comment (flavio) [reply src/testing-policies/01-intro.md:10]: Yeah, there could be more usages we can't even imagine right now

File: src/testing-policies/03-cluster-operators.md (hunk 2 above)
Comment (flavio) [reply src/testing-policies/03-cluster-operators.md:254]: Good point, I didn't know about that. I'll update also the other places where I linked this old repo

**Comment by flavio** [comment]
I've updated all the links to the bat project. If you are fine with that we can go ahead and merge this PR

**Pull request merged**